src/openforms/submissions/utils.py

```diff
@@ -335,3 +336,13 @@ def get_report_download_url(request: Request, report: SubmissionReport) -> str:
         kwargs={"report_id": report.id, "token": token},
     )
     return request.build_absolute_uri(download_url)
+
+
+def get_filtered_submission_admin_url(form_id: int, state: int, from_time: str) -> str:
+    query_params = {
+        "form__id__exact": form_id,
+        "needs_on_completion_retry__exact": state,
+        "from_time": from_time,
+    }
+    submissions_admin_url = furl(reverse("admin:submissions_submission_changelist"))
+    return submissions_admin_url.add(query_params).url
```

Comment:
the call-sites with positional arguments and "magic numbers" are not very readable, so let's enforce you have to use kwargs:

```suggestion
def get_filtered_submission_admin_url(form_id: int, *, filter_retry: bool, from_time: str) -> str:
    query_params = {
        "form__id__exact": form_id,
        "needs_on_completion_retry__exact": 1 if filter_retry else 0,
        "from_time": from_time,
    }
    submissions_admin_url = furl(reverse("admin:submissions_submission_changelist"))
    return submissions_admin_url.add(query_params).url
```